### PR TITLE
Read file

### PR DESCRIPTION
**Pull request by nadiaivc**
чо

### CODE REVIEW

File: Read_USB_512/Read_USB_512/Source.c [hunk 1]
```diff
@@ -40,21 +47,30 @@ struct MBH
...
+	unsigned char Empty[420];//90-509
 	unsigned char Signature[2];//510 - 511
 };
-	
+#pragma pack(pop)
+
+struct MBH MyMBH;
 
-void Read_file(char buffer[], int size)
+void Read_file()
```
Comment (useG) [inline Read_USB_512/Read_USB_512/Source.c]: Bad usage. What if we need to read second sector or 12312'th? And there will be another structure input or raw data?

File: Read_USB_512/Read_USB_512/Source.c [hunk 2]
```diff
@@ -40,21 +47,30 @@ struct MBH
...
+
+	ReadFile(handle, &MyMBH, sizeof(MyMBH), &end, NULL);
 	printf("Error %d.\n", GetLastError());
-	if (sizeof(My_MBH) != end) 
-		CloseHandle(handle);
 }
 
+int CheckMBH()
+{
+	if (MyMBH.Signature[0] != 0x55 && MyMBH.Signature[1] != 0xAA)
```
Comment (useG) [inline Read_USB_512/Read_USB_512/Source.c]: Good. Try to combine this two conditions into one (compare with 0x55AA).

File: Read_USB_512/Read_USB_512/Source.c [hunk 3]
```diff
@@ -40,21 +47,30 @@ struct MBH
...
+void Read_file()
 {
 	DWORD end;
-	struct MBH My_MBH;
-	ReadFile(handle, &My_MBH, size, &end, NULL);
+
+	ReadFile(handle, &MyMBH, sizeof(MyMBH), &end, NULL);
 	printf("Error %d.\n", GetLastError());
-	if (sizeof(My_MBH) != end) 
-		CloseHandle(handle);
```
Comment (useG) [inline Read_USB_512/Read_USB_512/Source.c]: Good that you check ReadFile for fail, but you have no appropriate action when ReadFile fails (CloseHandle inappropriate here)

File: Read_USB_512/Read_USB_512/Source.c [hunk 4]
```diff
@@ -40,21 +47,30 @@ struct MBH
...
+{
+	if (MyMBH.Signature[0] != 0x55 && MyMBH.Signature[1] != 0xAA)
+	{
+		printf("Error: no signature\n");
+		return EXIT_FAILURE;
+	}
+	return EXIT_SUCCESS;
+}
 
 int Create_file()
```
Comment (useG) [inline Read_USB_512/Read_USB_512/Source.c]: Try to unify Create_file independetly on USB letter path

File: Read_USB_512/Read_USB_512/Source.c [hunk 5]
```diff
@@ -72,9 +88,11 @@ int Create_file()
...
 {
-	unsigned char buffer[512];
-	if (Create_file() != EXIT_FAILURE)
-		Read_file(buffer, sizeof(buffer));
-	CloseHandle(handle);
-
+	if (Create_file() == EXIT_SUCCESS)
+	{
+		Read_file();
+		CloseHandle(handle);
```
Comment (useG) [inline Read_USB_512/Read_USB_512/Source.c]: If Read_file successed then CloseHandle should not take place here. Close handle at the end of the program or at fail

File: Read_USB_512/Read_USB_512/Source.c (hunk 2 above)
Comment (nadiaivc) [reply Read_USB_512/Read_USB_512/Source.c]: :c
image

Comment (useG) [reply Read_USB_512/Read_USB_512/Source.c]: Try to solve it by yourself, practise is the best teacher.

File: Read_USB_512/Read_USB_512/Source.c [hunk 6]
```diff
@@ -4,8 +4,11 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
+
 HANDLE handle;
-struct MBH
```
[the remark this answers is not in the document]
Comment (nadiaivc) [reply Read_USB_512/Read_USB_512/Source.c]: but it's an abbreviation(

File: Read_USB_512/Read_USB_512/Source.c [hunk 7]
```diff
@@ -24,11 +27,15 @@ struct MBH
...
+	struct BPB bpb;
 	unsigned int Number_of_BIOS_disk;//36-39//32-��������� ������ ����� ����� FAT (� ��������)
-	unsigned char Update_mode;//40-41//����� ���������� ���������� �������� FAT.\
+	unsigned char Update_mode[2];//40-41//����� ���������� ���������� �������� FAT.\
 								���� ��� 7 ����� 1, ������� ������ ���� ����� FAT, ������ ������� ������������ ��������� 0-3.\
 								� ��������� ������ ��� ��������� FAT �������� ����������� ������� ���� �����
-
 	unsigned char Name_version[2];//42-43//�������� � �������������� ����� ������
 	unsigned int Klaster;//44-47//�������, � ������� ��������� �������� �������
 	WORD FSINFO;//48-49//������, � ������� ��������� ��������� FSINFO
```
[the remark this answers is not in the document]
Comment (nadiaivc) [reply Read_USB_512/Read_USB_512/Source.c]: What can I do with an abbreviation?

**Pull request merged**